lib/IDE/Refactoring.cpp

```diff
@@ -1928,6 +1928,64 @@ bool RefactoringActionLocalizeString::performChange() {
   return false;
 }
 
+static std::unique_ptr<CharSourceRange>
+  findSourceRangeToWrapInCatch(ResolvedCursorInfo CursorInfo,
+                               SourceFile *TheFile,
+                               SourceManager &SM) {
+  Expr *E = CursorInfo.TrailingExpr;
+  if (!E)
+    return nullptr;
+  auto Node = ASTNode(E);
+  auto NodeChecker = [](ASTNode N) {
+    return N.isStmt(StmtKind::Brace);
+  };
+  ContextFinder Finder(*TheFile, Node, NodeChecker);
+  Finder.resolve();
+  auto Contexts = Finder.getContexts();
+  if (Contexts.size() == 0)
+    return nullptr;
+  auto TargetNode = Contexts.back();
+  Stmt *S = TargetNode.dyn_cast<Stmt*>();
+  BraceStmt *BStmt = dyn_cast<BraceStmt>(S);
+  auto ConvertToCharRange = [&SM](SourceRange SR) {
+    return Lexer::getCharSourceRangeFromSourceRange(SM, SR);
+  };
+  if (!BStmt)
+    return nullptr;
+  auto ExprRange = ConvertToCharRange(E->getSourceRange());
+  // Check elements of the deepest BraceStmt, pick one that covers expression.
+  for (auto Elem: BStmt->getElements()) {
+    auto ElemRange = ConvertToCharRange(Elem.getSourceRange());
+    if (ElemRange.contains(ExprRange))
+      TargetNode = Elem;
+  }
+  auto ResultRange = ConvertToCharRange(TargetNode.getSourceRange());
+  return llvm::make_unique<CharSourceRange>(ResultRange);
```

Comment:
We don't need to use unique pointer here; we can return `CharSourceRange` directly and check `isValid()` to determine if a range is found.